Doc/library/stdtypes.rst

```diff
@@ -4749,6 +4749,198 @@ define these methods must provide them as a normal Python accessible method.
 Compared to the overhead of setting up the runtime context, the overhead of a
 single class dictionary lookup is negligible.
 
+
+.. _types-genericalias:
+
+Generic Alias Type
+==================
+
+.. index::
+   object: GenericAlias
+   pair: Generic; Alias
+
+Most of the time, the :ref:`subscription expression <subscriptions>` calls the
+:term:`method` :meth:`__getitem__` of an :term:`object`. However, the
+subscription of a :term:`class` itself calls the :func:`classmethod
+<classmethod>` :meth:`__class_getitem__` of the class instead. The classmethod
+:meth:`__class_getitem__` returns a ``GenericAlias`` object.
+
+The ``GenericAlias`` object acts as a proxy for :term:`generic types
+<generic type>`, implementing *parameterized generics* - a specific instance
+of a generic with the types for container elements provided. It is intended
+primarily for type :term:`annotations <annotation>`.  ``GenericAlias`` objects
+are created by expressions like ``list[int]``.
+
+The type for the ``GenericAlias`` object is :data:`types.GenericAlias`.
+
+.. impl-detail::
+
+   The C equivalent of :meth:`__getitem__` - :c:func:`PyObject_GetItem` - is
+   called first for builtin classes and types. When :c:func:`PyObject_GetItem`
+   detects that it was called on a type, it then calls :meth:`__class_getitem__`
+   and returns the ``GenericAlias`` object.
+
+.. describe:: generic[X, Y, ...]
+
+   Defines a generic containing elements of types *X*, *Y*, and more depending
+   on the *generic* used.  For example, for a function expecting a :class:`list`
+   containing :class:`float` elements::
+
+      def average(values: list[float]) -> float:
+          return sum(values) / len(values)
+
+   Another example for :term:`mapping` objects, using a :class:`dict`.  In this
+   case, the function expects a ``dict`` for its ``body`` argument. The ``body``
+   has keys of type :class:`str` and their corresponding values are lists which
+   hold :class:`int` elements::
+
+      def send_post_request(url: str, body: dict[str, list[int]]) -> None:
+          ...
+
+The builtin functions :func:`isinstance` and :func:`issubclass` do not accept
+parameterized generic types for their second argument::
+
+   >>> isinstance([1, 2], list[str])
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   TypeError: isinstance() argument 2 cannot be a parameterized generic
+   >>>
+   >>> issubclass(list, list[str])
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   TypeError: issubclass() argument 2 cannot be a parameterized generic
+
+The Python runtime does not enforce type :term:`annotations <annotation>`.
```

Comment:
I'd include `type` in the label for the link, so it reads "...enforce **type annotations**" instead of "...enforce type **annotations**" (the latter misplaces the emphasis in the sentence).

Same for an earlier occurrence of the same phrase.